ONSdigital/census-rm-qid-batch-runner
# Rabbit Queue Backup & Restore

# Motivation and Context
We might have to rebuild the Rabbit cluster in-between the first stage of Census (print file generation) and later stages closer to the go-live date. In which case, it might be convenient for us to have a mechanism to store all the unconsumed messages sat on a queue, so that we can restore those messages to their queues at some future date.

# What has changed
Created two scripts:
- dump_queue_to_files.py
- dump_files_to_queue.py

# How to test?
Spin up a local environment. Load a queue with some messages. Look at the README section on how to run the scripts, then:
1. Run `dump_queue_to_files.py` with the appropriate parameters, and check the files are present and the queue is drained.
2. Run `dump_files_to_queue.py` with the appropriate parameters, and check that the messages are back on the queue.

# Links
Trello: https://trello.com/c/CdzlSkrf

# Screenshots (if appropriate):
